app/src/main/java/it/niedermann/owncloud/notes/persistence/NoteServerSyncHelper.java

```diff
@@ -364,13 +367,21 @@ private void pushLocalChanges() {
                 } catch (NextcloudHttpRequestFailedException e) {
                     if (e.getStatusCode() == 304) {
                         Log.d(TAG, "Server returned HTTP Status Code 304 - Not Modified");
+                        return LoginStatus.OK;
```

Comment:
What happens if there where multiple changes? `db.getLocalModifiedNotes` will return multiple notes and the for-loop is executed multiple times. However, in some cases, the execution is cancelled due to these new return statements. This means, all local changes that are behind a local change with error will not be executed.

An interesting use case is if the storage is full. I change a note and see the error message. Then I want to delete old notes in order to free some storage. However, the deletion will never be executed, since `pushLocalChanges` is cancelled after the first error.